File: app/modules/Environment.java
package modules;

import ui.PlayEnvironment;

/**
 * Created by Beemen on 30/07/2015.
 */
public class Environment {
    private static IEnvironment _Current;

    public static IEnvironment Current() {
        if(_Current == null) {
            _Current = new PlayEnvironment();
        }
        return _Current;
    }

}

File: Modules/Environment.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using end_user_gui.UI;

namespace end_user_gui.Modules
{
    public class Environment
    {
        private static IEnvironment _Current;

        public static IEnvironment Current()
        {
            if (_Current == null)
            {
                _Current = new PlayEnvironment();
            }
            return _Current;
        }

    }
}